synfig-studio/src/gui/workarea.cpp

```diff
@@ -1516,7 +1534,7 @@ WorkArea::on_drawing_area_event(GdkEvent *event)
 			double y(event->button.y), x(event->button.x);
 
 			// Erase the guides if dragged into the rulers
-			if((!std::isnan(x) && x<0.0) || (!std::isnan(y) && y<0.0))
+			if(((!std::isnan(x) && x<0.0) || (!std::isnan(y) && y<0.0)) && !get_lock_guides())
```

Comment:
If guide is not allowed to be dragged (i.e. it is locked), this situation won't ever be reached. So, no need to change this code here.